src/app/core/component.ts

```diff
@@ -44,12 +54,9 @@ export default class Component {
 
         // Create and set the ports here itself
 
-        const cleanparamdata = {};
-        for (const key in this.__params.parameters) {
-            cleanparamdata[key] = this.__params.parameters[key].getValue();
-        }
+        const cleanparamdata = this._params.parameters
 
-        const ports = Registry.featureSet.getComponentPorts(cleanparamdata, this.__type);
+        const ports = Registry.featureSet.getComponentPorts(cleanparamdata, this._type);
```

Comment:
Is this not giving you an error? and has the `getComponentPorts` been changed to work with this?